appIcons.js

```diff
@@ -166,6 +166,11 @@ const MyAppIcon = new Lang.Class({
 
     _updateRunningStyle: function() {
         this.parent();
+        if (this._dtdSettings.get_boolean('isolate-workspaces')) {
```

Comment:
I included your lines, they work better (totally missed the `dot.show()`). See patch [8c6f89738e13c14a3d3514af37afdfedbf32337a]

I have a further comment regarding `this._onFocusAppChanged()`. I actually moved it out of the `if`, so that it is always activated.

The problem doesn't show often, but it is present even if we don't use workspace isolation. I don't claim that this is the nicest solution, but for now it works.
I tested this on Gnome-Shell 3.20. If you use the built-in theme (so that the focus style is used) and you open a non-favorite app, focus is not shown (until you change to another window and back).
The same happens when using workspace-isolation and changing workspaces: non-favorite apps don't show focus at first.

Both problems are fixed when including `this._onFocusAppChanged()`. It could actually be moved to this line https://github.com/micheleg/dash-to-dock/blob/master/appIcons.js#L108. That way it would be clearer that we do it for onWindowsChanged.

What do you think?